docs/glossary.md

```diff
@@ -0,0 +1,53 @@
+# Terminology
+
+### Jobs 
+
+Jobs, or Primary Jobs, are special types of tasks run in simulations.
+
+### Tasks
+
+Tasks are any thing run within simulations.
+
+### Taskflow
+
+Taskflows are a collection of tasks and jobs which are run for simulations.
+
+### Workflow
+
+Workflows are collections of taskflows which define how you run simulations and jobs.
+
+### Projects
+
+Projects are collections of simulations.
+
+### Simulations
+
+Simulations are collections of workflows.
```

Comment:
By Gödel's ghost yes, it seems to be, or at least redundant. This could be mixed with my confusion on the definition of "Simulation" you brought up later.